### PR TITLE
Oracle module: check locking and dispute period

### PR DESCRIPTION
**Pull request by haythem96**
This PR add the functions to check if a locking or dispute period is over.
Should be reviewed/merged after #57 .

### CODE REVIEW

File: contracts/Oracle.sol [hunk 1]
```diff
@@ -3,10 +3,133 @@
...
+    }
+
+    /**
+     * @notice set oracle locking period
+     * @dev can only be called by owner
+     * @param _oracle oracle address
+     * @param _lockingPeriod locking period
+     */
+    function setLockingPeriod(address _oracle, uint256 _lockingPeriod) external onlyOwner {
+        oracleLockingPeriod[_oracle] = _lockingPeriod;
```
Comment (antoncoding) [inline contracts/Oracle.sol:123]: Should we have events for this and `setDisputePeriod`?

File: test/oracle.test.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
+import {ethers} from 'ethers'
```
Comment (antoncoding) [inline test/oracle.test.ts:1]: where do you use it?

File: test/oracle.test.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        'oracle dispute period mismatch',
+      )
+    })
+
+    it('should check if locking period is over when price timestmap equal to zero', async () => {
+      const isOver = await oracle.isDisputePeriodOver(batchOracle, await time.latest(), {from: owner})
+      const expectedResult = false
+      assert.equal(isOver, expectedResult, 'dispute period check mismatch')
+    })
+  })
```
Comment (antoncoding) [inline test/oracle.test.ts:83]: Add test for: 
* batchPrice.timestamp != 0 (https://coveralls.io/builds/32468982/source?filename=contracts/Oracle.sol#L99)

File: contracts/Oracle.sol [hunk 4]
```diff
@@ -3,10 +3,133 @@
...
+    /**
+     * @notice get batch price
+     * @param _batch batch hash
+     * @param _timestamp price timestamp
+     * @return price and timestap at which price submitted to this contract
+     */
+    function getBatchPrice(bytes32 _batch, uint256 _timestamp) public view returns (uint256, uint256) {
+        Price memory batchPrice = batchPriceAt[_batch][_timestamp];
+        return (batchPrice.price, batchPrice.timestamp);
+    }
```
Comment (antoncoding) [inline contracts/Oracle.sol:43]: Remove this function if we don't have setter and tests in this PR.

File: contracts/Oracle.sol (hunk 1 above)
Comment (haythem96) [reply contracts/Oracle.sol:123]: yeah will add events in the setter PR.

File: test/oracle.test.ts (hunk 3 above)
Comment (haythem96) [reply test/oracle.test.ts:83]: Will add it in another PR when I add the set price function.

File: contracts/Oracle.sol (hunk 4 above)
Comment (haythem96) [reply contracts/Oracle.sol:43]: This one was added in the getters PR, I think we can keep it as I will add tests for it in the next PR.

File: test/oracle.test.ts (hunk 2 above)
Comment (haythem96) [reply test/oracle.test.ts]: Yeah I installed it and didn't use it, was playing with it to mimic `keccak256(abi.encode())` hahsing as web3js is not great at that... I removed it for now.

File: contracts/Oracle.sol [hunk 5]
```diff
@@ -67,4 +77,72 @@ contract Oracle is Ownable {
     function getOracleDisputePeriod(address _oracle) public view returns (uint256) {
         return oracleDisputePeriod[_oracle];
     }
+
+    /**
+     * @notice check if locking period is over
+     * @param _batch batch hash
```
Comment (aparnakr) [inline contracts/Oracle.sol:83]: expand on what batch is

**Pull request merged**